### PR TITLE
fix: form autocomplete

### LINKED ISSUE #5913: 🔦 Feature: add autocomplete in Forms input fields
### Please confirm if feature request does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the usecase for the feature

I was thinking to use nocodb shared Form view for "subscribe to our Newsletter" page. But after a quick test I realized that even though I specified input as "email" there wasn't autocomplete="email" (https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) assigned, which would be really valuable for "subscribe to Newsletter" page

### Suggested Solution

either allow for user to set autocomplete attribute, or autodetect it and auto assign it for simple case (like email)

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change Summary

Closes #5913 

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/cell/Email.vue [hunk 1]
```diff
@@ -59,7 +70,10 @@ watch(
     v-if="editEnabled"
     :ref="focus"
     v-model="vModel"
-    class="w-full outline-none text-sm px-2"
+    class="w-full focus:!outline-none focus:ring-0 focus:border-none !border-none !outline-none text-sm px-2"
+    :type="isForm ? 'email' : 'text'"
+    :autocomplete="isForm ? 'email' : 'text'"
+    @blur="editEnabled = false"
```
Comment (DarkPhoenix2704) [inline packages/nc-gui/components/cell/Email.vue:76]:
```suggestion

```

File: packages/nc-gui/assets/style.scss [hunk 2]
```diff
@@ -587,3 +587,11 @@ input[type='number'] {
...
   @apply !block !py-1.5;
 }
+
+input[type='url'] {
+  @apply !p-0;
+}
+
+input[type='email'] {
+  @apply !p-0;
+}
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:597]: can be combined

File: packages/nc-gui/components/cell/Email.vue [hunk 3]
```diff
@@ -70,8 +73,10 @@ watch(
     v-if="editEnabled"
     :ref="focus"
     v-model="vModel"
-    class="w-full outline-none text-sm px-1 py-2"
     :placeholder="isEditColumn ? '(Optional)' : ''"
+    class="w-full focus:!outline-none focus:ring-0 focus:border-none !border-none !outline-none text-sm px-2"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Email.vue:77]: By adding `outline-none`, it should be enough for covering focus case as well. Hence, it seems `focus:!outline-none` is not necessary. When I inspect the input element, I didn't see the case that you need to use `!` to force override.

File: packages/nc-gui/components/cell/Email.vue [hunk 4]
```diff
@@ -70,8 +73,10 @@ watch(
...
+    :type="isForm ? 'email' : 'text'"
+    :autocomplete="isForm ? 'email' : 'text'"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Email.vue:79]: define a variable and use it here. don't do repetitive stuff.

File: packages/nc-gui/components/cell/Url.vue [hunk 5]
```diff
@@ -92,7 +95,9 @@ watch(
       :ref="focus"
       v-model="vModel"
       :placeholder="isEditColumn ? 'Enter default URL (Optional)' : ''"
-      class="outline-none text-sm w-full px-2 py-2 bg-transparent h-full"
+      class="outline-none focus:!outline-none focus:ring-0 focus:border-none !border-none !outline-none text-sm w-full px-2 py-2 bg-transparent h-full"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Url.vue:98]: same as above. double check if `!` is really required.

File: packages/nc-gui/assets/style.scss (hunk 2 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/assets/style.scss:597]: ok

File: packages/nc-gui/components/cell/Email.vue (hunk 3 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/cell/Email.vue:77]: updated

File: packages/nc-gui/components/cell/Email.vue (hunk 4 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/cell/Email.vue]: small typo. fixed

File: packages/nc-gui/assets/style.scss [hunk 6]
```diff
@@ -588,10 +588,6 @@ input[type='number'] {
   @apply !block !py-1.5;
 }
 
-input[type='url'] {
-  @apply !p-0;
-}
-
-input[type='email'] {
+input[type='url'] input[type='email'] {
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:599]: Missing a comma here

```css
input[type='url'],
input[type='email'] {
  @apply !p-0;
}
```

File: packages/nc-gui/components/cell/Email.vue [hunk 7]
```diff
@@ -74,9 +74,9 @@ watch(
     :ref="focus"
     v-model="vModel"
     :placeholder="isEditColumn ? '(Optional)' : ''"
-    class="w-full focus:!outline-none focus:ring-0 focus:border-none !border-none !outline-none text-sm px-2"
+    class="w-full focus:ring-0 focus:border-none border-none outline-none text-sm px-2"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Email.vue:77]: `focus:border-none` should be not necessary. please double check.

File: packages/nc-gui/components/cell/Url.vue [hunk 8]
```diff
@@ -60,6 +60,8 @@ const vModel = computed({
 
 const isValid = computed(() => value && isValidURL(value))
 
+const isInput
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Url.vue:63]: ??

File: packages/nc-gui/components/cell/Email.vue (hunk 7 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/cell/Email.vue:77]: ok

File: packages/nc-gui/components/cell/Email.vue [hunk 9]
```diff
@@ -70,8 +73,10 @@ watch(
     v-if="editEnabled"
     :ref="focus"
     v-model="vModel"
-    class="w-full outline-none text-sm px-1 py-2"
     :placeholder="isEditColumn ? $t('labels.optional') : ''"
+    class="w-full focus:!outline-none focus:ring-0 !border-none !outline-none text-sm px-2"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Email.vue:77]: I've left a comment on `!`. It seems the changes got overrode. Please double check.

File: packages/nc-gui/components/cell/Url.vue [hunk 10]
```diff
@@ -92,7 +95,9 @@ watch(
       :ref="focus"
       v-model="vModel"
       :placeholder="isEditColumn ? $t('labels.enterDefaultUrlOptional') : ''"
-      class="outline-none text-sm w-full px-2 py-2 bg-transparent h-full"
+      class="outline-none focus:!outline-none focus:ring-0 !border-none !outline-none text-sm w-full px-2 py-2 bg-transparent h-full"
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Url.vue:98]: I've left a comment on `!`. It seems the changes got overrode. Please double check.

**Pull request closed without merging**